### ISSUE TITLE
Paragraph maxIntrinsicWidth incorrect when text wraps

### ISSUE THREAD

**Issue by yjbanov** [flutter/flutter#21965]
## Steps to Reproduce

Flutter Engine computes incorrect `maxIntrinsicWidth` when `ParagraphConstraints` cause the paragraph to wrap. IOW, the following test should fail, but it passes:

```dart
  testWidgets('odd Ahem width', (WidgetTester tester) async {
    final ui.ParagraphBuilder builder = ui.ParagraphBuilder(ui.ParagraphStyle(
      fontFamily: 'Ahem',
      fontStyle: FontStyle.normal,
      fontWeight: FontWeight.normal,
      fontSize: 10.0,
    ));
    builder.addText('Test Ahem');
    final ui.Paragraph paragraph = builder.build();
    paragraph.layout(ui.ParagraphConstraints(width: 50.0));
    expect(paragraph.maxIntrinsicWidth, 80.0);

    paragraph.layout(ui.ParagraphConstraints(width: 1000.0));
    expect(paragraph.maxIntrinsicWidth, 90.0);
  });
```

Specifically, we should be getting `maxIntrinsicWidth` of 90.0 for both constraints. However, the engine incorrectly returns 80.0 for constraint 50.0.

/cc @Hixie @GaryQian

**Comment by Hixie** [comment]
I could also understand a result of 40.0 given a constraint of 50.0, though we'd have to update the docs to be clearer.

**Comment by mklim** [comment]
Huh. Looks like updating my fork also automatically applied the `Fixes flutter/flutter#21965` from one of the commits again for some reason. Reopening.

**Comment by GaryQian** [comment]
Framework tests have been updated during roll. This can be closed.
